src/proto/media/file/dir.rs

```diff
@@ -20,25 +20,20 @@ impl Directory {
         Self(RegularFile::new(handle))
     }
 
-    /// Read the next directory entry
+    /// Read the next directory entry.
     ///
-    /// Try to read the next directory entry into `buffer`. If the buffer is too small, report the
-    /// required buffer size as part of the error. If there are no more directory entries, return
-    /// an empty optional.
+    /// Under the hood, this wraps [`super::RegularFile::read`]. Please refer to it's documentation.
     ///
     /// The input buffer must be correctly aligned for a `FileInfo`. You can query the required
     /// alignment through the `Align` trait (`<FileInfo as Align>::alignment()`).
     ///
     /// # Arguments
     /// * `buffer`  The target buffer of the read operation
     ///
-    /// # Errors
-    /// * `uefi::Status::NO_MEDIA`           The device has no media
-    /// * `uefi::Status::DEVICE_ERROR`       The device reported an error, the file was deleted,
-    ///                                      or the end of the file was reached before the `read()`.
-    /// * `uefi::Status::VOLUME_CORRUPTED`   The filesystem structures are corrupted
-    /// * `uefi::Status::BUFFER_TOO_SMALL`   The buffer is too small to hold a directory entry,
-    ///                                      the required buffer size is provided into the error.
+    /// # Return Value
```

Comment:
I don't think having a separate `Return Value` section is standard; from looking around in the Rust std docs it seems like they usually describe the return value as part of the main body of the docs. A few random examples: https://doc.rust-lang.org/std/io/trait.Read.html#tymethod.read, https://doc.rust-lang.org/std/vec/struct.Vec.html#method.split_at_spare_mut, https://doc.rust-lang.org/std/process/struct.Command.html#method.spawn

Reply:
obsolete